packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L1StandardBridge.sol

```diff
@@ -0,0 +1,318 @@
+     *        length, these contracts provide no guarantees about its content.
+     */
+    function _initiateETHDeposit(
+        address _from,
+        address _to,
+        uint32 _l2Gas,
+        bytes memory _data
+    )
+        internal
+    {
+        // Construct calldata for finalizeDeposit call
+        bytes memory message =
+            abi.encodeWithSelector(
+                iOVM_L2ERC20Bridge.finalizeDeposit.selector,
+                0,
+                Lib_PredeployAddresses.OVM_ETH,
+                _from,
+                _to,
+                msg.value,
+                _data
+            );
+
+        // Send calldata into L2
+        sendCrossDomainMessage(
+            l2TokenBridge,
+            _l2Gas,
+            message
+        );
+
+        emit ETHDepositInitiated(_from, _to, msg.value, _data);
+    }
+
+    /**
+     * @inheritdoc iOVM_L1ERC20Bridge
+     */
+    function depositERC20(
+        address _l1Token,
+        address _l2Token,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        external
+        override
+        virtual
+        onlyEOA()
+    {
+        _initiateERC20Deposit(_l1Token, _l2Token, msg.sender, msg.sender, _amount, _l2Gas, _data);
+    }
+
+     /**
+     * @inheritdoc iOVM_L1ERC20Bridge
+     */
+    function depositERC20To(
+        address _l1Token,
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        external
+        override
+        virtual
+    {
+        _initiateERC20Deposit(_l1Token, _l2Token, msg.sender, _to, _amount, _l2Gas, _data);
+    }
+
+    /**
+     * @dev Performs the logic for deposits by informing the L2 Deposited Token
+     * contract of the deposit and calling a handler to lock the L1 funds. (e.g. transferFrom)
+     *
+     * @param _l1Token Address of the L1 ERC20 we are depositing
+     * @param _l2Token Address of the L1 respective L2 ERC20
+     * @param _from Account to pull the deposit from on L1
+     * @param _to Account to give the deposit to on L2
+     * @param _amount Amount of the ERC20 to deposit.
+     * @param _l2Gas Gas limit required to complete the deposit on L2.
+     * @param _data Optional data to forward to L2. This data is provided
+     *        solely as a convenience for external contracts. Aside from enforcing a maximum
+     *        length, these contracts provide no guarantees about its content.
+     */
+    function _initiateERC20Deposit(
+        address _l1Token,
+        address _l2Token,
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        internal
+    {
+        // When a deposit is initiated on L1, the L1 Bridge transfers the funds to itself for future withdrawals.
+        IERC20(_l1Token).safeTransferFrom(
```

Comment:
Needs a test that shows this will revert when `_l1Token == address(0)` (+ a comment probably)